rest/audit_test.go

```diff
@@ -46,58 +52,163 @@ func TestAuditInjectableHeader(t *testing.T) {
+					"cid":         auditFieldValueIgnored,
+					"real_userid": map[string]any{"domain": "sgw", "user": "alice"},
+					"extra":       "field",
+				},
+			},
+		},
+		{
+			name: "rejected public request",
+			auditableAction: func(t testing.TB) {
+				RequireStatus(t, rt.SendUserRequest(http.MethodGet, "/db/", "", "incorrect"), http.StatusUnauthorized)
+			},
+			expectedAuditEventFields: map[base.AuditID]base.AuditFields{
+				base.AuditIDPublicUserAuthenticationFailed: {
+					"cid":         auditFieldValueIgnored,
+					"auth_method": "basic",
+					"db":          "db",
+					"username":    "incorrect",
+				},
+			},
+		},
+		{
+			name: "anon admin request",
+			auditableAction: func(t testing.TB) {
+				RequireStatus(t, rt.SendAdminRequest(http.MethodGet, "/db/", ""), http.StatusOK)
+			},
+			expectedAuditEventFields: map[base.AuditID]base.AuditFields{
+				// TODO: Admin auth event
+				//base.AuditIDAdminUserAuthenticated: {
+				//	"cid":         auditFieldValueIgnored,
+				//},
+				base.AuditIDReadDatabase: {
+					"cid": auditFieldValueIgnored,
+				},
+			},
 		},
-		*/
-		/* FIXME: metrics requests are not currently audited
 		{
-			name: "metrics request",
-			fn:   rt.SendMetricsRequestWithHeaders,
+			name: "authed admin request",
+			auditableAction: func(t testing.TB) {
+				if base.UnitTestUrlIsWalrus() {
+					// Skip this subtest if running with walrus - it has no support for admin auth
+					t.Skip("Skipping test that requires admin auth - Walrus not supported")
+				}
+				RequireStatus(t, rt.SendAdminRequestWithAuth(http.MethodGet, "/db/", "", base.TestClusterUsername(), base.TestClusterPassword()), http.StatusOK)
+			},
+			expectedAuditEventFields: map[base.AuditID]base.AuditFields{
+				// TODO: Admin auth event
+				//base.AuditIDAdminUserAuthenticated: {
+				//	"cid":         auditFieldValueIgnored,
+				//	"real_userid": map[string]any{"domain": "cbs", "user": base.TestClusterUsername()},
+				//},
+				base.AuditIDReadDatabase: {
+					"cid":         auditFieldValueIgnored,
+					"real_userid": map[string]any{"domain": "cbs", "user": base.TestClusterUsername()},
+				},
+			},
 		},
-		*/
 	}
 	for _, testCase := range testCases {
-		headers := map[string]string{
-			headerName:      `{"extra":"field"}`,
-			"Authorization": getBasicAuthHeader("alice", RestTesterDefaultUserPassword),
-		}
 		t.Run(testCase.name, func(t *testing.T) {
-			output := base.AuditLogContents(t, func() {
-				// FIXME: this test only works with /db because not all http requests are logged
-				RequireStatus(t, testCase.fn(http.MethodGet, "/db/", "", headers), http.StatusOK)
-			})
-			events := bytes.Split(output, []byte("\n"))
-			foundEvent := false
-			for _, rawEvent := range events {
-				if bytes.TrimSpace(rawEvent) == nil {
+			output := base.AuditLogContents(t, testCase.auditableAction)
+			events := jsonLines(t, output)
+
+			// for each event, check the fields match what we expected
```

Comment:
do we want to assert that specific events occur only once? I am not sure.

Reply:
I'll do that also as a guard to ensure future subtest writers don't include multiple auditable actions in one `AuditLogContents`